homeworks/franktoth/6_testing/test_bankaccount.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+from bankaccount import BankAccount
+
+
+# Fixture to create BankAccounts with different initial balances
+@pytest.fixture(scope="class")
+def accounts():
+    account1 = BankAccount("Frank", 500)
+    account2 = BankAccount("Steve", 700)
+    invalid_account = BankAccount("Account Error", -250)
+    return account1, account2, invalid_account
+
+
+class TestBankAccount:
+    # Test deposit with various inputs using parametrize
+    @pytest.mark.parametrize("deposit_amount, expected_balance, expected_exception", [
+        (50, 550, None), 
+        (0, 330, None), 
+        (-30, 100, ValueError)
+    ])
+    def test_deposit(self, accounts, deposit_amount, expected_balance, expected_exception):
```

Comment:
ezt a tesztet is szedd szét nyugodtan, legyen egy teszted pl test_deposit_with_invalid_values() ahol megnézed hogy raise-el e exception-öket.
Aztán lehet egy külön teszted pl_ test_deposit_success() vagy valami hasonló ahol azt nézed hogy a balance értéke úgy növekedik e ahogy te arra számítasz.